### ISSUE TITLE
Global shortcut to hide/unhide Helium

### ISSUE THREAD

**Issue by harrisonw1** [JadenGeller/Helium#71]

**Comment by JadenGeller** [comment]
Definitely a cool idea. Planned! :dancer:

**Comment by fabersky** [comment]
I added a boss-alert :smile: https://github.com/JadenGeller/Helium/pull/75
@JadenGeller what do you think about it?

**Comment by ryh** [comment]
`⌘+H` couldn't  do this feature?

**Comment by fabersky** [comment]
Yes, but when you do ⌘+H, then it's not that easy to show Helium again!

**Comment by JadenGeller** [comment]
You can't use Command+H when Helium isn't frontmost. What is desired is a global shortcut (that will work in any other app and hopefully not clash with other shortcuts) that will always hide Helium. It's unclear that a global shortcut to show Helium is necessary as that isn't enough of an emergency to warrant a shortcut (maybe it would be reasonable to make the hide shortcut, when already hidden, show Helium).

**Comment by ryh** [comment]
So this app is for people watch video in work time and can hide it quickly from boss? :no_good:

**Comment by JadenGeller** [comment]
Definitely not, but it could be useful to hide it via a keyboard shortcut nevertheless.

**Comment by fabersky** [comment]
No @ryh , I don't think so because you aren't doing something illegal! 
But I think it could be helpful in emergency situations! It should be named "girlfriend alert" too  :metal:

**Comment by cobalamin** [comment]
+1: This is actually what I thought would be the central feature of this 😄
And it's not just useful for watching Netflix during work or what have you - in fact, I think of this feature being a boon to productivity.

I found this while searching for something like the overlays TotalTerminal/TotalFinder provide, but with a browser. Those things are extremely useful! You can show and hide them globally whenever you need them, they auto-hide on unfocus like normal windows, but you can pin them if you need them while working in another window simultaneously.

Example: I'm looking for something on StackOverflow and found a nice, but complex answer for my problem, so I want to try it out in an editor while being able to very quickly go back to the answer in my browser.

**Comment by fabersky** [comment]
From what I can see, @JadenGeller left the development. Hey J!!

**Comment by micnguyen** [comment]
+1 for this feature.

**Comment by JonasDoebertin** [comment]
👍 Would be quite useful to be able to hide Helium at any times with a keyboard shortcut.

**Comment by jameslafa** [comment]
Here is the temporary solution:

Open Automator, select to create a Service, configure to have it receive no input in any application.

From the library, double-click Utilities » Run AppleScript and enter the following into the large text area:

```
on run {input, parameters}
	
	tell application "System Events"
		set names to name of application processes
		if names contains "Helium" then
			tell application process "Helium"
				if visible then
					tell application "Helium" to activate
					key code 49
					set visible to false
				else
					set visible to true
					tell application "Helium" to activate
					key code 49
				end if
			end tell
		else
			display dialog "Helium is not running"
		end if
	end tell
	
	return input
end run
```

Save under any name. Assign a keyboard shortcut in System Preferences » Keyboard » Keyboard Shortcuts » Services.

It does the job :)

**Comment by avncharlie** [comment]
If anyone uses BetterTouchTool, it's also pretty easy to make an AppleScript that toggles hiding/unhiding Helium and binding that to anything you want. It would be a great feature to have built into the app though.
